firstuseauthenticator/firstuseauthenticator.py

```diff
@@ -147,11 +147,21 @@ def authenticate(self, handler, data):
         password = data['password']
         # Don't enforce password length requirement on existing users, since that can
         # lock users out of their hubs.
-        if not self._validate_password(password) and not self._user_exists(username):
-            handler.custom_login_error = (
-                'Password too short! Please choose a password at least %d characters long.'
-                % self.min_password_length
-            )
+        with dbm.open(self.dbm_path, 'c', 0o600) as db:
+            stored_pw = db.get(username.encode(), None)
+            #check if the user exists and there is no password for the user
+            if self._user_exists(username) and stored_pw is None:
```

Comment:
There's no need to check if the user exists in the database, because we also want to ensure a minimum password length for when the user doesn't exist and gets created by the authenticator.

Because `stored_pw` will be `None` when the user doesn't have a password set, or when it isn't in the database, this is the only check we need to make.

Reply:
The workflow would then just be:
`if stored_pw is None`  and `if password introduced doesn't validate` -> `throw error`

Reply:
Yes, Makes Sense to me.
I believe there is no other way, an user can be added apart from admin adding him/her. If there is any, we need to accomodate that kind of cases too (ignoring the fresh registration done by user where we are already taking care about this constraint)